### PR TITLE
nrf5340/nrf52xxx: Add support for enabling instruction/data cache in hal_system_init() and add dependencies for BUS DRIVER

### PR DESCRIPTION
**Pull request by vrahane**
- nrf5340: Add support for enabling instruction/data cache in hal_system_init() based on syscfg and add dependencies for BUS_DRIVER

- nrf52xxx: Enable instruction code cache in hal_system_init() based on syscfg

### CODE REVIEW

**Comment by andrzej-kaczmarek** [comment]
split this into two separate PRs or at least do not mix unrelated bus driver and cache changes in single commit

**Comment by sjanc** [comment]
why not just always enable icache than?

**Comment by vrahane** [comment]
We can enable cache always but the problem with that is that sometimes the performance actually degrades instead of becoming better. It should be left to the developer to decide if he/she wants to enable it or not. Another thing is, if someone wants to monitor the performance of the cache, example: hit/miss counts, they would enable the cache and look at those registers, so, keeping it conditional is better.

**Comment by vrahane** [comment]
@andrzej-kaczmarek yes agreed, will add a separate commit for it.

**Comment by vrahane** [comment]
@andrzej-kaczmarek added a separate commit for the BUS driver changes.

File: hw/mcu/nordic/nrf5340/src/hal_system.c [hunk 1]
```diff
@@ -72,6 +81,7 @@ hal_debugger_connected(void)
 void
 hal_system_clock_start(void)
 {
+
```
Comment (sjanc) [inline hw/mcu/nordic/nrf5340/src/hal_system.c:84]: nitpick

File: hw/mcu/nordic/nrf5340/src/hal_system.c [hunk 2]
```diff
@@ -36,6 +36,15 @@
 void
 hal_system_init(void)
 {
+
```
Comment (sjanc) [inline hw/mcu/nordic/nrf5340/src/hal_system.c:39]: why empty line?

**Comment by utzig** [comment]
@vrahane The Travis fail on the branch is a bit weird, did you first push the branch upstream instead of to your own fork?

**Comment by sjanc** [comment]
fair enough, two nitpicks though

File: hw/mcu/nordic/nrf5340/src/hal_system.c (hunk 1 above)
Comment (vrahane) [reply hw/mcu/nordic/nrf5340/src/hal_system.c]: done

File: hw/mcu/nordic/nrf5340/src/hal_system.c (hunk 2 above)
Comment (vrahane) [reply hw/mcu/nordic/nrf5340/src/hal_system.c]: removed

**Comment by vrahane** [comment]
@utzig your are right, I did push to origin by mistake and then deleted it right after. The recent commits should fix that I guess.

**Comment by utzig** [comment]
> @utzig your are right, I did push to origin by mistake and then deleted it right after. The recent commits should fix that I guess.

Looks like it did, seems to require a rebase to sync with Travis again, thanks!

**Comment by vrahane** [comment]
@utzig done!

**Review by sjanc — APPROVED** [review]

**Pull request merged**